Source/Moya+Internal.swift

```diff
@@ -174,17 +174,17 @@ private extension MoyaProvider {
     }
 
     func sendUploadFile(_ target: Target, request: URLRequest, queue: DispatchQueue?, file: URL, progress: ProgressBlock? = nil, completion: @escaping Completion) -> CancellableToken {
-        let alamoRequest = manager.upload(file, with: request)
+        let alamoRequest = target.validate ? manager.upload(file, with: request).validate() : manager.upload(file, with: request)
```

Comment:
Although I am not sure about naming, you might want to do this step in two lines. Something like

```swift
let initialRequest = manager.upload(file, with: request)
let alamoRequest = target.validate ? initialRequest.validate() : initialRequest
```

So we can prevent code duplication and possible errors due to that in the future. What do you think?